pkg/storage/pluginstore/grpc.go

```diff
@@ -297,13 +298,20 @@ func ConvertFloatToInt(src interface{}) interface{} {
 		}
 		return tv
 	case []interface{}:
+		toBson := make(bson.D, 0)
```

Comment:
Let's size this based on the length of tv
```suggestion
		toBson := make(bson.D, len(tv))
```